doc/build/kconfig/setting.rst

```diff
@@ -259,9 +259,24 @@ symbol properties, so the above ``default`` is equivalent to
       	...
       	depends on DEP2
 
+   Symbols without any listed dependencies are treated as having a dependency
+   on the "y" symbol, which is always set. Applying the ORing rule results in
+   the definitions below resolving to a symbol with no dependencies. If ``BAR``
+   is set, ``FOO`` will be enabled regardless of the value of ``DEP1``.
+
+   .. code-block:: none
+
+      config FOO
+      	...
+      	depends on DEP1
+
+      config FOO
+      	...
+      	default y if BAR
+
    When making changes to :file:`Kconfig.defconfig` files, always check the
    symbol's direct dependencies in one of the :ref:`interactive configuration
-   interfaces <menuconfig>` afterwards. It is often necessary to repeat
+   interfaces <menuconfig>` afterwards. It is **required** to repeat
```

Comment:
not accepted.

Often a downstream project actually want to weaken a symbol, like this:
```
# Kconfig.zephyr

config FOO
    bool "Foo"
    depends on BAR
```

but downstream has an alternative feature:
```
# Kconfig.downstream

config FOO
    depends on DOWN_BAR
```

and that's a perfectly valid use-case.
But stating required can mislead the reader into believing it is required to duplicate all dependencies.

Also in a situation where the repeated symbol is guarded by `if`, then it is also not necessary to repeat the `depends on`.

Like:
```
# Kconfig.A

config FOO
    depends on BAR
```
and then having Kconfig.B which sources a Kconfig.C which changes the default, like:
```
# Kconfig.B

if BAR

source Kconfig.C

endif
```
and then

```
# Kconfig.C

config FOO
    default y
```
in Kconfig.C there is no need for repeating the `depends on` due to the `if BAR` before the source line.